cmd/titus-storage/ebs.go

```diff
@@ -0,0 +1,312 @@
+			return err
+		}
+	} else {
+		return fmt.Errorf("Command '%s' unsupported. Must be either start or stop", command)
+	}
+	return nil
+}
+
+func ebsStart(ctx context.Context, ec2Client *ec2.EC2, c MountConfig) error {
+	var err error
+	l := logger.GetLogger(ctx)
+
+	ec2DeviceName, ok := getDeviceName(c.ebsVolumeID)
+	if ok {
+		l.Printf("EBS volume %s is already attached locally, moving on", c.ebsVolumeID)
+	} else {
+		err = waitForVolumeToBeUnused(ctx, c.ebsVolumeID, ec2Client, 30)
+		if err != nil {
+			return fmt.Errorf("%s is not unused. Not safe to detach. %s", c.ebsVolumeID, err)
+		}
+		ec2DeviceName, err = attachEBS(ctx, c.ebsVolumeID, ec2Client, c.instanceID)
+		if err != nil {
+			return err
+		}
+	}
+
+	device, err := GetActualBlockDeviceName(ec2DeviceName)
+	if err != nil {
+		return err
+	}
+
+	err = mkfsIfNeeded(device, c.ebsFStype)
+	if err != nil {
+		return err
+	}
+
+	err = mountIt(ctx, device, c.ebsFStype, c.ebsMountPoint, c.ebsMountPerm, c.pid1Dir)
+	if err != nil {
+		return err
+	}
+	l.Printf("finished setting up EBS %s at %s inside the container", c.ebsVolumeID, c.ebsMountPoint)
+	return nil
+}
+
+func ebsStop(ctx context.Context, ec2Client *ec2.EC2, c MountConfig) error {
+	l := logger.GetLogger(ctx)
+	ec2Device, ok := getDeviceName(c.ebsVolumeID)
+	if !ok {
+		l.Printf("EBS volume %s doesn't look attached. Nothing to do", c.ebsVolumeID)
+		return nil
+	}
+
+	actualDeviceName, err := GetActualBlockDeviceName(ec2Device)
+	if err != nil {
+		return err
+	}
+
+	// Sometimes ebsStop gets called *before* a container dies, so we need to wait.
+	// Sometimes ebsStop gets called *after* a container dies, that is fine, will move on quickly
+	// *Sometimes* ebsStop gets called *defore* a container dies, but *after* a *different* container
+	// comes up, see it is attached, and mounts. In this case we need to still panic here, but it is "ok"
+	// In that situation we should not continue to detach a volume out from underneath it
+	l.Printf("Waiting up to %d seconds for %s to be not in use...", 30, actualDeviceName)
+	err = waitForDeviceToBeNotInUse(actualDeviceName, 30)
+	if err != nil {
+		return fmt.Errorf("Not detaching volume: %s", err)
+	}
+
+	// By the time we get here, we are confident the volume is attached, not in use, and safe to detatch
+	err = detatchEBS(ctx, c.ebsVolumeID, ec2Client, c.instanceID)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+func getEc2Client() *ec2.EC2 {
+	sess := session.Must(session.NewSessionWithOptions(session.Options{
+		SharedConfigState: session.SharedConfigEnable,
+	}))
+	creds := credentials.NewCredentials(
+		&ec2rolecreds.EC2RoleProvider{
+			Client:       ec2metadata.New(sess),
+			ExpiryWindow: 5 * time.Minute,
+		},
+	)
+	sess.Config.Credentials = creds
+	*sess.Config.Region = os.Getenv("EC2_REGION")
+	return ec2.New(sess)
+}
+
+func waitForVolumeToBeUnused(ctx context.Context, ebsVolumeID string, ec2Client *ec2.EC2, timeout int) error {
+	l := logger.GetLogger(ctx)
```

Comment:
In general, the "easier" pattern here is:

```go
logger.G(ctx).Info(...)
```

As opposed to getting the logger and using it by hand